File: patch_api.py
import re

with open('src/lib/api.ts', 'r') as f:
    content = f.read()

# Update Presencas select
content = content.replace(
    ".select(`*, funcionario:funcionarios!inner(*, funcao:funcoes(*), obra:obras(*))`)\n      .or",
    ".select(`*, funcionario:funcionarios!inner(*, funcao:funcoes(*), obra:obras(*))`)\n      .or"
)

# Update getDashboardStats logic for meia_diaria
dashboard_logic = """
    presencasHojeData?.forEach(p => {
      if (p.presente) {
        presentesHoje++;
        let valor = Number((p.funcionario as any)?.funcao?.valor_diaria || 0);
        if (p.meia_diaria) valor = valor / 2;
        valorTotalHoje += valor;
      } else {
"""
content = re.sub(
    r'presencasHojeData\?\.forEach\(p => \{\s*if \(p\.presente\) \{\s*presentesHoje\+\+;\s*valorTotalHoje \+= Number\(\(p\.funcionario as any\)\?\.funcao\?\.valor_diaria \|\| 0\);\s*\} else \{',
    dashboard_logic.strip(),
    content
)

with open('src/lib/api.ts', 'w') as f:
    f.write(content)

File: patch_api.js
const fs = require('fs');
let code = fs.readFileSync('src/lib/api.ts', 'utf8');

const targetFunction = `  updateFuncionario: async (
    id: string,
    funcionario: Partial<Funcionario>,
  ): Promise<Funcionario> => {
    if (!supabase) throw new Error("Supabase não configurado");
    const { data, error } = await withEmpresa(supabase.from("funcionarios").update(funcionario),
    )
      .eq("id", id)
      .select()
      .single();
    if (error) throw error;
    return data as any;
  },`;

const newFunction = \`  updateFuncionario: async (
    id: string,
    funcionario: Partial<Funcionario>,
  ): Promise<Funcionario> => {
    if (!supabase) throw new Error("Supabase não configurado");
    const { data, error } = await withEmpresa(supabase.from("funcionarios").update(funcionario),
    )
      .eq("id", id)
      .select()
      .single();
    if (error) throw error;
    return data as any;
  },
  updateFuncionariosObra: async (ids: string[], obraId: string): Promise<void> => {
    if (!supabase) throw new Error("Supabase não configurado");
    const { error } = await withEmpresa(supabase.from("funcionarios").update({ obra_id: obraId }).in("id", ids));
    if (error) throw error;
  },\`;

code = code.replace(targetFunction, newFunction);
fs.writeFileSync('src/lib/api.ts', code);
